src/modules/hub.bicep

```diff
@@ -131,6 +145,207 @@ resource uploadSettingsJson 'Microsoft.Resources/deploymentScripts@2020-10-01' =
   }
 }
 
+// Key Vault for storing secrets
+
+module keyVault 'Microsoft.KeyVault/vaults/deploy.bicep' = {
+  name: 'keyVault'
+  dependsOn: [
+    dataFactory
+  ]
+  params: {
+    name: keyVaultName
+    location: location
+    tags: resourceTags
+    enablePurgeProtection: false
+    accessPolicies: [
+      {
+        objectId: dataFactory.outputs.systemAssignedPrincipalId
+        tenantId: subscription().tenantId
+        permissions: {
+          secrets: [
+            'get'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+module keyVaultSecret 'secrets/storageaccountsecret.bicep' = {
+  name: 'keyVaultSecret'
+  dependsOn: [
+    storageAccount
+    keyVault
+  ]
+  params: {
+    keyVaultName: keyVaultName
+    storageAccountName: storageAccountName
+    secretName: storageAccountName
+  }
+}
+
+// Azure Data Factory and Pipelines
+module dataFactory 'Microsoft.DataFactory/factories/deploy.bicep' = {
+  name: 'dataFactory'
+  params: {
+    name: dataFactoryName
+    location: location
+    tags: resourceTags
+    systemAssignedIdentity: true
+  }
+}
+
+module linkedService_keyVault 'linkedservices/linkedservice.bicep' = {
+  name: 'linkedService_keyVault'
+  dependsOn: [
+    dataFactory
+    keyVaultSecret
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: keyVaultName
+    linkedServiceType: 'AzureKeyVault'
+    linkedServiceTypeProperties:  {
+      baseUrl: keyVaultUri
+    }
+  }
+}
+
+module linkedService_storageAccount 'linkedservices/linkedservice.bicep' = {
+  name: 'linkedService_storageAccount'
+  dependsOn: [
+    dataFactory
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    linkedServiceType: 'AzureBlobFS'
+    linkedServiceTypeProperties: {
+      url: reference('Microsoft.Storage/storageAccounts/${storageAccountName}', '2019-04-01').primaryEndpoints.dfs
+      accountKey: {
+        type: 'AzureKeyVaultSecret'
+        store: {
+          referenceName: keyVaultName
+          type: 'LinkedServiceReference'
+        }
+        secretName: storageAccountName
+      }
+    }
+  }
+}
+
+module msexports 'datasets/adlsv2.bicep' = {
+  name: 'msexports'
+  dependsOn: [
+    linkedService_storageAccount
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    compressionCodec: 'none'
+    datasetType: 'DelimitedText'
+    containerName: exportContainer.name
+  }
+}
+
+module ingestion_csv 'datasets/adlsv2.bicep' = {
+  name: 'ingestion_csv'
```

Comment:
In the call you mentioned we'd only support one format. Should we only create one dataset or do you want to eventually support multiple?

Reply:
I want to support both parquet and .csv.gz as the output.
Parquet will be the only path that's enabled, but by modifying the trigger one can toggle from parquet to .csv.gz output.
Or remove the delete action from the end of the csv.gz code path and have that execute before the parquet conversion so we can output both.